### PR TITLE
Run integration tests in Github Workflows

### PR DESCRIPTION
**Pull request by Kulezi**
The goal of this PR is to run integration tests on the cluster described by docker-compose.yml in GH workflows.
It extends the Makefile with ability to run integration tests in docker, this is used by the modified workflow.

This PR also resolves a small bug in OpenShardConn, when context is done, instead of immediate return from the function, it would try all ports as connections wouldn't open due to context being done, bloating the integration test log with thousands of lines.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -57,6 +57,16 @@ else
...
 
+.PHONY: docker-integration-test
+docker-integration-test: RUN=Integration
+docker-integration-test:
+	@CGO_ENABLED=0 GOOS=linux go test -v -tags integration -c -o ./integration-test.dev $(PKG)
+	@docker run --name "integration-test" \
+		--network scylla_go_driver_public \
+		-v "$(PWD)/testdata:/testdata" \
+		-v "$(PWD)/integration-test.dev:/usr/bin/integration-test:ro" \
+		-i --read-only --rm ubuntu integration-test -test.v -test.run $(RUN) $(ARGS)
```
Comment (piodul) [inline Makefile:68]: Why is `testdata/testdata`, mounted here, but in the `pkg-integration-test` target it's not?

File: transport/conn.go [hunk 2]
```diff
@@ -376,6 +376,10 @@ func OpenShardConn(ctx context.Context, addr string, si ShardInfo, cfg ConnConfi
 	it := ShardPortIterator(si)
 	maxTries := (maxPort-minPort+1)/int(si.NrShards) + 1
 	for i := 0; i < maxTries; i++ {
+		if err := ctx.Err(); err != nil {
+			return nil, fmt.Errorf("aborted opening connection on shard %d: %w", si.Shard, err)
+		}
+
```
Comment (piodul) [inline transport/conn.go:382]: The fix this commit introduces doesn't seem to be related to the main purpose of the PR (adding integration tests to the CI). It should be sent in a separate PR.

File: .github/workflows/go.yml [hunk 3]
```diff
@@ -28,6 +30,21 @@ jobs:
...
+      - name: Ensure nodes are up
+        run: |
+          until docker compose exec node1 nodetool status; do sleep 1; done
+          while [[ $(docker compose exec node1 nodetool status | grep -c ^UN) != "3" ]]; do sleep 1; done
+
+      - name: Run integration tests
+        run: make docker-integration-test
+        
+      - name: Stop cluster
+        run: GITHUB_WORKSPACE=${{ github.workspace }} docker-compose down
```
Comment (piodul) [inline .github/workflows/go.yml:50]: What is the purpose for the `GITHUB_WORKSPACE` env variable here?

File: Makefile (hunk 1 above)
Comment (Kulezi) [reply Makefile:69]: Missed this as workflow is being ran on ubuntu not mac, fixed.

File: transport/conn.go (hunk 2 above)
Comment (Kulezi) [reply transport/conn.go]: Will move that to #271 as its related to logs, dropped

File: .github/workflows/go.yml (hunk 3 above)
Comment (Kulezi) [reply .github/workflows/go.yml]: It's unused, removed.

File: Makefile [hunk 4]
```diff
@@ -51,12 +51,23 @@ else ifeq ($(OS),Darwin)
 	@CGO_ENABLED=0 GOOS=linux go test -v -tags integration -c -o ./integration-test.dev $(PKG)
 	@docker run --name "integration-test" \
 		--network scylla_go_driver_public \
+		-v "$(PWD)/testdata:/testdata" \
```
Comment (piodul) [inline Makefile:54]: Nit: it would be preferable to fix this in a separate commit. The current commit is about the `docker-integration-test`, so this change seems unrelated.

**Review by piodul — APPROVED** [review]
